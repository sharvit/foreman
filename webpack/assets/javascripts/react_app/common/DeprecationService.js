/* eslint-disable no-console, max-len */

export const deprecate = (oldMethod, newMethod, version = '1.22') =>
  console.warn(
    `DEPRECATION WARNING: you are using deprecated ${oldMethod}, it will be removed in Foreman ${version}. Use ${newMethod} instead.`
  );

export const deprecateObjectProperty = (
  obj,
  oldProp,
  newProp,
  version = '1.20'
) => {
  const oldPropPointer = obj[oldProp];

  Object.defineProperty(obj, oldProp, {
    get: () => {
      deprecate(oldProp, newProp, version);
      return oldPropPointer;
    },
  });
};
