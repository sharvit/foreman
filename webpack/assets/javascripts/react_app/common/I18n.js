import Jed from 'jed';
import { addLocaleData } from 'react-intl';
import forceSingleton from './forceSingleton';
import {
  loadVendorIntl,
  loadVendorIntlLocale,
  loadReactIntlLocale,
} from 'vendor-intl';

class IntlLoader {
  constructor(locale, timezone) {
    this.fallbackIntl = !global.Intl;

    [this.locale] = locale.split('-');
    this.timezone = this.fallbackIntl ? 'UTC' : timezone;
    this.ready = this.init();
  }

  async init() {
    await this.fetchIntl();
    const { default: localeData } = await loadReactIntlLocale(this.locale);

    addLocaleData(localeData);
    return true;
  }

  async fetchIntl() {
    if (this.fallbackIntl) {
      const { default: Intl } = await loadVendorIntl();

      global.Intl = Intl;
      await loadVendorIntlLocale(this.locale);
    }
  }
}

const [htmlElemnt] = document.getElementsByTagName('html');
const langAttr = htmlElemnt.getAttribute('lang') || 'en';
const timezoneAttr = htmlElemnt.getAttribute('data-timezone') || 'UTC';

export const intl = forceSingleton(
  'Intl',
  () => new IntlLoader(langAttr, timezoneAttr)
);

const cheveronPrefix = () => (window.I18N_MARK ? '\u00BB' : '');
const cheveronSuffix = () => (window.I18N_MARK ? '\u00AB' : '');

export const documentLocale = () => langAttr;

const getLocaleData = () => {
  const locales = window.locales || {};
  const locale = documentLocale().replace(/-/g, '_');

  if (locales[locale] === undefined) {
    // eslint-disable-next-line no-console
    console.log(
      `could not load translations for ${locale} locale, falling back to default locale.`
    );
    return { domain: 'app', locale_data: { app: { '': {} } } };
  }

  return locales[locale];
};

export const jed = forceSingleton('Jed', () => new Jed(getLocaleData()));

export const translate = (...args) =>
  `${cheveronPrefix()}${jed.gettext(...args)}${cheveronSuffix()}`;
export const ngettext = (...args) =>
  `${cheveronPrefix()}${jed.ngettext(...args)}${cheveronSuffix()}`;

export const { sprintf } = jed;

const i18n = {
  translate,
  ngettext,
  jed,
  sprintf,
  intl,
};
export default i18n;

window.__ = translate;
window.n__ = ngettext;
