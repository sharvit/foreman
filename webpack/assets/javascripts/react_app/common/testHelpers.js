export default {
  mockStorage: () => {
    const storage = {};

    return {
      setItem: (key, value) => {
        storage[key] = value || '';
      },
      getItem: key => (key in storage ? storage[key] : null),
      removeItem: key => {
        delete storage[key];
      },
      get length() {
        return Object.keys(storage).length;
      },
      key: i => {
        const keys = Object.keys(storage);

        return keys[i] || null;
      },
    };
  },
};

// a helper method for invoking a class method (for unit tests)
// obj = a class
// func = a tested function
// objThis = an object's this
// arg = function args

export const classFunctionUnitTest = (obj, func, objThis, args) =>
  obj.prototype[func].apply(objThis, args);
