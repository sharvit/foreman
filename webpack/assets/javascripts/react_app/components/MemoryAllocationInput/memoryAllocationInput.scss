
.help-block-icon {
  margin-right: 3px;
}

.common-numberInput {
  .react-number-input {
    display: flex !important;

    input {
      flex: 1;
      height: 28px;
    }
  }
}
