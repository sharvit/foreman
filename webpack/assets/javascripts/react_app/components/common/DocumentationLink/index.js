import React from '@theforeman/vendor/react';
import PropTypes from '@theforeman/vendor/prop-types';
import { MenuItem, Icon } from '@theforeman/vendor/patternfly-react';
import { newWindowOnClick } from '../../../common/helpers';
import { translate as __ } from '../../../../react_app/common/I18n';

const DocumentationLink = ({ href, children }) => (
  <MenuItem key="documentationUrl" href={href} onClick={newWindowOnClick(href)}>
    <Icon type="fa" name="question-circle" />
    {` ${children}`}
  </MenuItem>
);

DocumentationLink.propTypes = {
  href: PropTypes.string.isRequired,
  children: PropTypes.node,
};

DocumentationLink.defaultProps = {
  children: __('Documentation'),
};

export default DocumentationLink;
