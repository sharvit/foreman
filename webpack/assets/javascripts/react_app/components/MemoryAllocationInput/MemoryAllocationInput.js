import React, { useCallback, useState } from 'react';
import PropTypes from 'prop-types';
import { WarningTriangleIcon, ErrorCircleOIcon } from '@patternfly/react-icons';
import { HelpBlock, FormGroup, Grid, Col, Row } from 'patternfly-react';
import { translate as __ } from '../../common/I18n';
import NumberInput from '../common/forms/NumberInput';
import './memoryAllocationInput.scss';
import { noop } from '../../common/helpers';

const MemoryAllocationInput = ({
  label,
  defaultValue,
  onChange,
  maxValue,
  maxValueSizeType,
  recommendedMaxValue,
  recommendedMaxValueSizeType,
}) => {
  const GBFormat = 'GB';
  const MBFormat = 'MB';
  const GBStep = 1;
  const MBStep = 256;
  const [value, setValue] = useState(defaultValue);
  const [sizeType, setSizeType] = useState(MBFormat);
  const [step, setStep] = useState(MBStep);

  const getValidationState = useCallback(() => {
    if (value > maxValue && maxValueSizeType === sizeType && maxValue != null)
      return 'error';
    else if (
      value > recommendedMaxValue &&
      recommendedMaxValueSizeType === sizeType &&
      recommendedMaxValue != null
    )
      return 'warning';
    return null;
  }, [recommendedMaxValue, maxValue, value]);

  const handleChange = v => {
    if (value === 1 && sizeType === GBFormat && v < value) {
      setValue(768);
      setSizeType(MBFormat);
      setStep(MBStep);
    } else if (value === 768 && sizeType === MBFormat && v > value) {
      setValue(1);
      setSizeType(GBFormat);
      setStep(GBStep);
    } else if (value === 1 && sizeType === MBFormat && v > value) {
      setValue(256);
    } else {
      setValue(v);
    }
    onChange(value);
  };
  const validationState = getValidationState();
  return (
    <FormGroup validationState={validationState}>
      <Grid>
        <Row>
          <Col>
            <NumberInput
              value={value}
              format={v => `${v} ${sizeType}`}
              step={step}
              onChange={v => handleChange(v)}
              label={label}
            />
          </Col>
        </Row>
        {validationState === 'warning' && (
          <HelpBlock>
            <WarningTriangleIcon className="help-block-icon" />
            {__('Specified value is higher than recommended maximum')}
          </HelpBlock>
        )}
        {validationState === 'error' && (
          <HelpBlock>
            <ErrorCircleOIcon className="help-block-icon" />
            {__('Specified value is higher than maximum value')}
          </HelpBlock>
        )}
      </Grid>
    </FormGroup>
  );
};

MemoryAllocationInput.propTypes = {
  /** Set the label of the memory allocation input */
  label: NumberInput.propTypes.label,
  /** Set the default value of the numeric input */
  defaultValue: PropTypes.number,
  /** Set the recommended max value of the numeric input */
  recommendedMaxValue: PropTypes.number,
  /** Set the max value of the numeric input */
  maxValue: PropTypes.number,
  /** Set the max value size type of the numeric input */
  maxValueSizeType: PropTypes.oneOf(['MB', 'GB']),
  /** Set the recommended max value size type of the numeric input */
  recommendedMaxValueSizeType: PropTypes.oneOf(['MB', 'GB']),
  /** Set the onChange function of the numeric input */
  onChange: PropTypes.func,
};

MemoryAllocationInput.defaultProps = {
  label: __('Memory'),
  defaultValue: 1,
  onChange: noop,
  recommendedMaxValue: null,
  maxValue: null,
  maxValueSizeType: 'GB',
  recommendedMaxValueSizeType: 'GB',
};

export default MemoryAllocationInput;
