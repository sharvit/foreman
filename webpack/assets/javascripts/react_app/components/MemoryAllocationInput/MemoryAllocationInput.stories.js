import React from 'react';
import { number, action, text } from '@theforeman/stories';
import MemoryAllocationInput from './MemoryAllocationInput';

export default {
  title: 'Components|Form/MemoryAllocationInput',
  component: MemoryAllocationInput,
  parameters: {
    centered: { disable: true },
  },
};

export const UseMemoryAllocationInput = () => (
  <MemoryAllocationInput
    label={text('Label', 'Memory')}
    defaultValue={number('DefaultValue', 1)}
    onChange={action('Value was changed')}
    recommendedMaxValue={number('RecommendedMaxValue', 10)}
    recommendedMaxValueSizeType={text('RecommendedMaxValueSizeType', 'GB')}
    maxValue={number('MaxValue', 20)}
    maxValueSizeType={text('MaxValueSizeType', 'GB')}
  />
);
