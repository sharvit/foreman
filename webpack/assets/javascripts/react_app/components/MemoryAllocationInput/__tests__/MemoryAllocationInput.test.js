import { testComponentSnapshotsWithFixtures } from '../../../common/testHelpers';
import MemoryAllocationInput from '../MemoryAllocationInput';

const props = {
  label: 'Memory',
};

const fixtures = {
  'should render with default props': {
    ...props,
  },
};

describe('MemoryAllocationInput', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(MemoryAllocationInput, fixtures));
});
