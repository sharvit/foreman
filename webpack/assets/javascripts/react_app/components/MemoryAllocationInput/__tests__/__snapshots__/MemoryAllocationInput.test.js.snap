// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemoryAllocationInput rendering should render with default props 1`] = `
<FormGroup
  bsClass="form-group"
  validationState={null}
>
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
      >
        <NumberInput
          className=""
          disabled={false}
          format={[Function]}
          label="Memory"
          minValue={0}
          onChange={[Function]}
          precision={0}
          readOnly={false}
          step={256}
          value={1}
        />
      </Col>
    </Row>
  </Grid>
</FormGroup>
`;
